src/gui/torrentcontenttreeview.cpp

```diff
@@ -37,17 +37,31 @@
 #include <QTableView>
 #include <QThread>
 
+#include "base/bittorrent/abstractfilestorage.h"
 #include "base/bittorrent/common.h"
 #include "base/bittorrent/session.h"
 #include "base/bittorrent/torrenthandle.h"
 #include "base/bittorrent/torrentinfo.h"
+#include "base/exceptions.h"
 #include "base/global.h"
 #include "base/utils/fs.h"
 #include "autoexpandabledialog.h"
 #include "raisedmessagebox.h"
 #include "torrentcontentfiltermodel.h"
 #include "torrentcontentmodelitem.h"
 
+namespace
+{
+    QString getFullPath(const QModelIndex &idx)
+    {
+        const QModelIndex &parentIdx = idx.parent();
```

Comment:
`QModelIndex::parent()` return a value, not a reference.

Reply:
However, this is valid/well-formed code. Besides, I just moved it from somewhere.
But I'll change it anyway, since you don't like it.